### DISCUSSION TITLE
Getting odd MIME type check error

### DISCUSSION THREAD

**Discussion by Isaac-Tait** [vercel/next.js#68257] [Help] [answered]
### Summary

My site was running fine then I updated a page  (client requested wording nothing else) and now I am getting this error message along with a site loading with zero styling: 

`Refused to execute script from 'myURL/_next/static/dU0z1bvnjWYdHW3cbAer8/_ssgManifest.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.`

I cleared it the first time by running another build after clearing the cache on the Netlify end. Then a few days later the same error cropped up again (a site with no css and the MIME error message). 

Seems like #64350 is having similar issues but only in Lighthouse. 

Was there a change in NextJS that is making my site's css no longer discoverable in the directory assets?

### Additional information

```js
`Refused` to execute script from 'myURL/_next/static/dU0z1bvnjWYdHW3cbAer8/_ssgManifest.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.`
```

**Comment by icyJoseph** [comment]
I remember someone who had a script, or plugin, which accidentally, transformed CSS files into JavaScript, and IIRC, it would've also done that to `JSON` files. Let me search around.

**Reply by icyJoseph** [reply]
So here, https://github.com/vercel/next.js/issues/66221#issuecomment-2133037208 they had some special webpack config, that was causing CSS to end up as JS scripts. Could it be somewhat similar to your case?

**Comment by Joakim-Andersson** [comment]
I'm having this same exact issue on a new site. Everything works fine in dev mode, and the site was working fine all week in prod until I updated a small CSS file yesterday. Now, I get 404 & MIME type error.  Have you found a solution yet?

**Reply by Isaac-Tait** [reply]
@Joakim-Andersson When the error first appeared I ran another build after clearing the cache. That fixed it for a day or so. When it happened again I tried updating the outdated dependencies  (all the updates were minor - see code snippet below) and that has so far fixed the error 🤞🏾

Like you I pushed a small css (Tailwind to be precise) update to the site and somehow that caused the error to occur.

```
@headlessui/react 2.0.4 > 2.1.2
algoliasearch: 4.23.3 > ^4.24.0,
graphql: 16.8.2 > ^16.9.0,
instantsearch.css: 8.2.1 > ^8.3.0,
next: 14.2.4 > 14.2.5
react-instantsearch: 7.11.2 > 7.12.2,
tailwindcss: 3.4.4 > 3.4.7
```

EDIT: To note the second time the MIME error occurred my client told me and I had to visit about a dozen pages on the site before the error triggered (using a private window in a browser with the cache cleared btw so it wasn't the cache loading the CSS for a couple pages)...

**Reply by Joakim-Andersson** [reply]
Thanks! 
I just updated my dependencies and redeployed the site. It worked for about one hour, and then the CSS files went "missing" again. Very frustrating and not sure what to do haha

**Reply by Isaac-Tait** [reply]
Ugg that is frustrating... I'm worried that is going to pop back up for me too. So, I am definitely invested in trying to figure out what is going on.

Are you using app router or page router?
TailwindCSS or something else?

Could you copy and paste your package.json (mine is below - so we can compare)?

```
{
  "name": "page_router",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@headlessui/react": "^2.1.2",
    "@heroicons/react": "^1.0.6",
    "@tailwindcss/typography": "^0.5.13",
    "algoliasearch": "^4.24.0",
    "graphql": "^16.9.0",
    "instantsearch.css": "^8.3.0",
    "next": "14.2.5",
    "react": "^18",
    "react-dom": "^18",
    "react-instantsearch": "^7.12.2",
    "react-instantsearch-dom": "^6.40.4",
    "react-instantsearch-router-nextjs": "^7.12.2",
    "tailwindcss-hero-patterns": "^0.1.2"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.19",
    "eslint": "^8.57.0",
    "eslint-config-next": "14.2.5",
    "postcss": "^8",
    "tailwindcss": "^3.4.7"
  },
  "comments": {
    "dependencies": "@heroicons/react should not be upgraded to v2 or else it will cause the mobile menu to return an error"
  }
}
```

**Reply by icyJoseph** [reply]
When this error happens, is it a 404 that's returned for the script? with a full HTML page on it?

**Reply by icyJoseph** [reply, marked as answer]
Found this thread elsewhere, https://answers.netlify.com/t/intermittent-404-errors-and-inconsistent-page-loads-on-next-js-app-hosted-on-netlify/96997/12 any change it is somewhat related?

**Reply by Isaac-Tait** [reply]
My app is hosted on Netlify... and yes it was throwing a 404 error in conjunction with the MIME error. Maybe @Joakim-Andersson can upload a screenshot? My app isn't currently erroring out. 

Great sleuthing @icyJoseph 🫡

**Reply by Isaac-Tait** [reply]
That issue has since been closed (13 hours ago as of this posting) and re-directed to this one:
https://answers.netlify.com/t/javascript-bundle-issues-netlify-build-breaking-several-hours-after-deployment-completes/122901/2

**Reply by Isaac-Tait** [reply]
It might be resolved - Netlify released a patch. If you redeploy and purge the cache you should be good to go 🤞🏼
https://answers.netlify.com/t/javascript-bundle-issues-netlify-build-breaking-several-hours-after-deployment-completes/122901/50?u=isaac-tait

**Reply by ALFAMAS** [reply]
is there any fix
cause i am running on latest version but mines still facing same issue with CI/CD

**Comment by umutsesen** [comment]
getting this error on a just basic portfolio website, no solutions.. package/json: {
  "name": "umut-portfolio",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev --turbopack",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@microsoft/clarity": "^1.0.0",
    "i18next": "^25.3.2",
    "i18next-browser-languagedetector": "^8.2.0",
    "next": "15.4.1",
    "next-seo": "^6.8.0",
    "next-sitemap": "^4.2.3",
    "react": "19.1.0",
    "react-dom": "19.1.0",
    "react-i18next": "^15.6.0",
    "react-icons": "^5.5.0"
  },
  "devDependencies": {
    "@tailwindcss/postcss": "^4",
    "@types/node": "^20",
    "@types/react": "^19",
    "@types/react-dom": "^19",
    "tailwindcss": "^4",
    "typescript": "^5"
  }
}
